mediapipe/tasks/ios/test/text/text_classifier/MPPTextClassifierTests.m

```diff
@@ -0,0 +1,281 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "mediapipe/tasks/ios/common/sources/MPPCommon.h"
+#import "mediapipe/tasks/ios/text/text_classifier/sources/MPPTextClassifier.h"
+
+static NSString *const kBertTextClassifierModelName = @"bert_text_classifier";
+static NSString *const kRegexTextClassifierModelName =
+    @"test_model_text_classifier_with_regex_tokenizer";
+static NSString *const kNegativeText = @"unflinchingly bleak and desperate";
+static NSString *const kPositiveText = @"it's a charming and often affecting journey";
+static NSString *const kExpectedErrorDomain = @"com.google.mediapipe.tasks";
+
+#define AssertEqualErrors(error, expectedError)                                               \
+  XCTAssertNotNil(error);                                                                     \
+  XCTAssertEqualObjects(error.domain, expectedError.domain);                                  \
+  XCTAssertEqual(error.code, expectedError.code);                                             \
+  XCTAssertNotEqual(                                                                          \
+      [error.localizedDescription rangeOfString:expectedError.localizedDescription].location, \
+      NSNotFound)
+
+#define AssertEqualCategoryArrays(categories, expectedCategories)                          \
+  XCTAssertEqual(categories.count, expectedCategories.count);                              \
+  for (int i = 0; i < categories.count; i++) {                                             \
+    XCTAssertEqual(categories[i].index, expectedCategories[i].index);                      \
+    XCTAssertEqualWithAccuracy(categories[i].score, expectedCategories[i].score, 1e-6);    \
+    XCTAssertEqualObjects(categories[i].categoryName, expectedCategories[i].categoryName); \
+    XCTAssertEqualObjects(categories[i].displayName, expectedCategories[i].displayName);   \
```

Comment:
Would it be possible to add the index for these to the failure description?

Reply:
I have added for both swift and obj c